### PR TITLE
Propagate loadBalancerAddress

### PR DESCRIPTION
**Pull request by tokle**
@bjorncs fyi

### CODE REVIEW

File: config-model-api/src/main/java/com/yahoo/config/model/api/ModelContext.java [hunk 1]
```diff
@@ -41,6 +42,7 @@
         boolean multitenant();
         ApplicationId applicationId();
         List<ConfigServerSpec> configServerSpecs();
+        URI loadBalancerAddress();
```
Comment (bjorncs) [inline config-model-api/src/main/java/com/yahoo/config/model/api/ModelContext.java]: Note @tokle: The `AthenzIdentityProvider` assumes that the load balancer address is a hostname (no scheme or port). We'll have to update that component at the same time of merging this PR.

**Comment by bjorncs** [comment]
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:testCompile (default-testCompile) on project config-model: Compilation failure: Compilation failure:
> [ERROR] /source/config-model/src/test/java/com/yahoo/config/model/MockModelContext.java:[81,33] <anonymous com.yahoo.config.model.MockModelContext$1> is not abstract and does not override abstract method loadBalancerAddress() in com.yahoo.config.model.api.ModelContext.Properties
> [ERROR] /source/config-model/src/test/java/com/yahoo/vespa/model/VespaModelFactoryTest.java:[164,41] <anonymous com.yahoo.vespa.model.VespaModelFactoryTest$2$1> is not abstract and does not override abstract method loadBalancerAddress() in com.yahoo.config.model.api.ModelContext.Properties
> [ERROR] -> [Help 1]

File: config-model/src/main/java/com/yahoo/config/model/deploy/DeployProperties.java [hunk 2]
```diff
@@ -2,8 +2,11 @@
 package com.yahoo.config.model.deploy;
 
 import com.yahoo.config.model.api.ConfigServerSpec;
-import com.yahoo.config.provision.*;
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/config/model/deploy/DeployProperties.java]: Nice to see someone got their IntelliJ settiings for imports right.

File: config-model/src/main/java/com/yahoo/config/model/deploy/DeployProperties.java [hunk 3]
```diff
@@ -18,14 +21,16 @@
...
+    private final URI loadBalancerAddress;
     private final boolean hostedVespa;
     private final Version vespaVersion;
     private final Zone zone;
 
     private DeployProperties(boolean multitenant,
                              ApplicationId applicationId,
                              List<ConfigServerSpec> configServerSpecs,
-                             boolean hostedVespa, Version vespaVersion, Zone zone) {
+                             URI loadBalancerAddress, boolean hostedVespa, Version vespaVersion, Zone zone) {
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/config/model/deploy/DeployProperties.java]: It looks strange to have a list of params on one line compared to the above.

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java [hunk 4]
```diff
@@ -93,6 +96,8 @@
     private static final String xmlRendererId = RendererRegistry.xmlRendererId.getName();
     private static final String jsonRendererId = RendererRegistry.jsonRendererId.getName();
 
+    private static final Logger logger = Logger.getLogger(ContainerCluster.class.getName());
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java]: Did you use the wrong classname here?

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java [hunk 5]
```diff
@@ -162,7 +167,7 @@ private void addClusterContent(ContainerCluster cluster, Element spec, ConfigMod
         addLegacyFilters(spec, cluster);  // TODO: Remove for Vespa 7
 
         // Athenz copper argos
-        addIdentity(spec, cluster, context.getDeployState().getProperties().configServerSpecs());
+        addIdentity(spec, cluster, context.getDeployState().getProperties());
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java]: It's a bit ugly to propagate the full deploy properties. Please limit the scope of the arguments as much as possible.

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java [hunk 6]
```diff
@@ -690,20 +695,21 @@ public static void validateAndAddConfiguredComponents(ContainerCluster cluster,
...
     }
 
-    private void addIdentity(Element element, ContainerCluster cluster, List<ConfigServerSpec> configServerSpecs) {
+    private void addIdentity(Element element, ContainerCluster cluster, DeployProperties deployProperties) {
         Element identityElement = XML.getChild(element, "identity");
         if(identityElement != null) {
             String domain = XML.getValue(XML.getChild(identityElement, "domain"));
             String service = XML.getValue(XML.getChild(identityElement, "service"));
 
+            // Log to see if propagated correctly
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java]: Add a "TODO: remove" comment?

**Review by gjoranv — CHANGES_REQUESTED** [review]

File: config-model-api/src/main/java/com/yahoo/config/model/api/ModelContext.java (hunk 1 above)
Comment (tokle) [reply config-model-api/src/main/java/com/yahoo/config/model/api/ModelContext.java]: This PR terminates just before building the model. Will add a todo

File: config-model/src/main/java/com/yahoo/config/model/deploy/DeployProperties.java (hunk 3 above)
Comment (tokle) [reply config-model/src/main/java/com/yahoo/config/model/deploy/DeployProperties.java]: fixed

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java (hunk 4 above)
Comment (tokle) [reply config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java]: yes.... :(

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java (hunk 5 above)
Comment (tokle) [reply config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java]: fixed

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java (hunk 6 above)
Comment (tokle) [reply config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java]: done

**Comment by tokle** [comment]
@gjoranv I've fixed the build and addressed the comments. ptal

**Comment by bjorncs** [comment]
@tokle Merge conflict btw.

**Review by gjoranv — APPROVED** [review]

**Comment by gjoranv** [comment]
👍  Merge at will

**Pull request merged**